sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/BlobDecryptionPolicy.java

```diff
@@ -304,51 +352,121 @@ Flux<ByteBuffer> decryptBlob(String encryptedDataString, Flux<ByteBuffer> encryp
         });
     }
 
-    /**
-     * Gets and validates {@link EncryptionData} from a Blob's metadata
-     *
-     * @param encryptedDataString {@code String} of encrypted metadata
-     * @return {@link EncryptionData}
-     */
-    private EncryptionData getAndValidateEncryptionData(String encryptedDataString) {
-        if (encryptedDataString == null) {
-            if (requiresEncryption) {
-                throw LOGGER.logExceptionAsError(new IllegalStateException("'requiresEncryption' set to true but "
-                    + "downloaded data is not encrypted."));
-            }
-            return null;
+    private Flux<ByteBuffer> decryptV2(Flux<ByteBuffer> encryptedFlux, EncryptionData encryptionData,
+        byte[] contentEncryptionKey) {
+        // Buffer an exact region with the nonce and tag
+        final int gcmEncryptionRegionLength = encryptionData.getEncryptedRegionInfo().getDataLength();
+        final int nonceLength = encryptionData.getEncryptedRegionInfo().getNonceLength();
+        BufferStagingArea stagingArea =
+            new BufferStagingArea(gcmEncryptionRegionLength + TAG_LENGTH + nonceLength,
+                gcmEncryptionRegionLength + TAG_LENGTH + nonceLength);
+
+        return UploadUtils.chunkSource(encryptedFlux,
+                new com.azure.storage.common.ParallelTransferOptions()
+                    .setBlockSizeLong((long) gcmEncryptionRegionLength
+                        + TAG_LENGTH + nonceLength))
+            .flatMapSequential(stagingArea::write)
+            .concatWith(Flux.defer(stagingArea::flush))
+            .flatMapSequential(aggregator -> {
+                // Get the IV out of the beginning of the aggregator
+                byte[] gmcIv = aggregator.getFirstNBytes(nonceLength);
+
+                Cipher gmcCipher;
+                try {
+                    gmcCipher = getCipher(contentEncryptionKey, encryptionData, gmcIv, false);
+                } catch (InvalidKeyException e) {
+                    return Mono.error(LOGGER.logExceptionAsError(Exceptions.propagate(e)));
+                }
+
+                ByteBuffer decryptedRegion = ByteBuffer.allocate(gcmEncryptionRegionLength);
+                return aggregator.asFlux()
+                    .map(buffer -> {
+                        // Write into the preallocated buffer and always return this buffer.
+                        try {
+                            gmcCipher.update(buffer, decryptedRegion);
+                        } catch (ShortBufferException e) {
+                            return Mono.error(LOGGER.logExceptionAsError(Exceptions.propagate(e)));
```

Comment:
This should throw directly as this is in a `map` call and not a `flatMap` call and I think would result in `Mono<Mono<Exception>>`